modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -498,6 +498,27 @@ impl ClientModule for WalletClientModule {
         Some(self.cfg().fee_consensus.peg_out_abs)
     }
 
+    async fn handle_rpc(
+        &self,
+        method: String,
+        request: serde_json::Value,
+    ) -> BoxStream<'_, anyhow::Result<serde_json::Value>> {
+        Box::pin(try_stream! {
+            if method.as_str() == "get_wallet_summary" {
+                let _req: WalletSummaryRequest = serde_json::from_value(request)?;
+                let wallet_summary = self.get_wallet_summary()
+                    .await
+                    .map_err(|e| anyhow::anyhow!("Failed to fetch wallet summary: {}", e))?;
+                let result = serde_json::to_value(&wallet_summary)
+                    .map_err(|e| anyhow::anyhow!("Serialization error: {}", e))?;
+                yield result;
+            } else {
+                Err(anyhow::format_err!("Unknown method: {}", method))?;
+                unreachable!()
```

Comment:
This ready rather oddly, do you need that `unreachable!` macro there? Can't you just `return` the error?